apps/agentstack-server/src/agentstack_server/service_layer/services/files.py

```diff
@@ -54,21 +57,31 @@ async def extract_text(self, file_id: UUID, job_id: str):
             file = await uow.files.get(file_id=file_id)
             error_log.append(file.model_dump())
             user = await uow.users.get(user_id=file.created_by)
-            extraction.set_started(job_id=job_id)
+
+            extraction.set_started(
+                job_id=job_id,
+                backend=self._extraction_backend.backend_name,
+            )
             await uow.files.update_extraction(extraction=extraction)
             await uow.commit()
         try:
             file_url = await self._object_storage.get_file_url(file_id=file_id)
             error_log.append(f"file url: {file_url}")
-            async with self._extraction_backend.extract_text(file_url=file_url) as extracted_file:
-                extracted_db_file = await self.upload_file(
-                    file=extracted_file,
-                    user=user,
-                    file_type=FileType.EXTRACTED_TEXT,
-                    context_id=file.context_id,
-                    parent_file_id=file_id,
-                )
-            extraction.set_completed(extracted_file_id=extracted_db_file.id)
+            async with self._extraction_backend.extract_text(
+                file_url=file_url,
+                settings=extraction.extraction_metadata.settings if extraction.extraction_metadata else None,
+            ) as extracted_files_with_format:
+                extracted_files = []
+                for async_file, extraction_format in extracted_files_with_format:
+                    extracted_db_file = await self.upload_file(
```

Comment:
nitpick: could be done in parralel in an `asyncio.TaskGroup`

If you fix the streams in docling backend and we have one stream that is waiting for the other to finish on `await` I'm not sure if it won't break or timeout